clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -407,6 +407,16 @@ private void maybeRecordTimePerConnection(KafkaChannel channel, long startTimeNa
             channel.addNetworkThreadTimeNanos(time.nanoseconds() - startTimeNanos);
     }
 
+    private void prepare(KafkaChannel channel) throws IOException {
+        try {
+            channel.prepare();
+        } catch (IOException e) {
+            log.warn("Connection with {} terminated during authentication. This may indicate " +
+                    "authentication failure due to invalid credentials", channel.socketDescription());
```

Comment:
Nit: would it read better if we said "This may indicate that authentication failed due to invalid credentials"?